### PR TITLE
Simplify the way to pass the glyph drawing instructions from the worker to the main thread

### PR DESCRIPTION
**Pull request by calixteman**
and remove the use of eval in the font loader.

### CODE REVIEW

**Review by Snuffleupagus — APPROVED** [review]

**Comment by Rob--W** [comment]
This PR is associated with the fix for [CVE-2024-4367 ](https://github.com/mozilla/pdf.js/security/advisories/GHSA-wgrm-67xf-hhpq).

The advisory currently mentions `<= 4.1.392`, but in practice the version ranges are different, the precise details are below. Despite the presence of a version range that is unaffected by this specific bug, I still recommend using the latest versions only (v4.2.67+) because these older versions are also affected by another vulnerability (CVE-2018-5158; I've filed a PR to update the right metadata in Github's advisory database at https://github.com/github/advisory-database/pull/4456).

PDF.js versions and statuses:
- [v4.2.67](https://github.com/mozilla/pdf.js/releases/tag/v4.2.67) (released April 29, 2024) **unaffected** (fixed)
  - Commit: https://github.com/mozilla/pdf.js/commit/85e64b5c16c9aaef738f421733c12911a441cec6
  - Pull request: https://github.com/mozilla/pdf.js/pull/18015
- [v4.1.392](https://github.com/mozilla/pdf.js/releases/tag/v4.1.392) (released April 11, 2024) affected (release before this bug was fixed)
- [v1.10.88](https://github.com/mozilla/pdf.js/releases/tag/v1.10.88) (released Oct 27, 2017) affected (re-introduces the security vulnerability due to a typo fix)
  - Commit: https://github.com/mozilla/pdf.js/commit/ab4d5be19294e3dc2972efd07d1c1373cc8e7e85
  - Pull request: https://github.com/mozilla/pdf.js/pull/9008
- [v1.9.426](https://github.com/mozilla/pdf.js/releases/tag/v1.9.426) (released Aug 15, 2017) **unaffected** (release before the next affected version)
- [v1.5.188](https://github.com/mozilla/pdf.js/releases/tag/v1.5.188) (released Apr 21, 2016) **unaffected** (mitigated the security vulnerability by an accidental typo)
  - commit: https://github.com/mozilla/pdf.js/commit/1d12aed5cac6a0caccb0550e0aad4ef9f6bb263d
  - Pull request: https://github.com/mozilla/pdf.js/pull/7126 - the typo is later fixed by PR 9008.
- [v1.4.20](https://github.com/mozilla/pdf.js/releases/tag/v1.4.20) (released Jan 27, 2016) affected (release before the next release that accidentally fixed the vulnerable code)
- [v0.8.1181](https://github.com/mozilla/pdf.js/releases/tag/v0.8.1181) (released Apr 10, 2014) affected (first public release of PDF.js)
  - Commit: https://github.com/mozilla/pdf.js/commit/0e133f00902b9fef8422a932385a61a38ca85545
  - Pull request: https://github.com/mozilla/pdf.js/pull/3195

**Comment by zlm15122306538** [comment]
If the version is not upgraded, set isEvalSupported: false in which file

**Comment by zlm15122306538** [comment]
> and remove the use of eval in the font loader.

If the version is not upgraded, set isEvalSupported: false in which file

File: src/display/font_loader.js [hunk 1]
```diff
@@ -440,35 +434,85 @@ class FontFaceObject {
...
 
-    // If we can, compile cmds into JS for MAXIMUM SPEED...
-    if (this.isEvalSupported && FeatureTest.isEvalSupported) {
-      const jsBuf = [];
-      for (const current of cmds) {
-        const args = current.args !== undefined ? current.args.join(",") : "";
-        jsBuf.push("c.", current.cmd, "(", args, ");\n");
+    const commands = [];
+    for (let i = 0, ii = cmds.length; i < ii; ) {
+      switch (cmds[i++]) {
```
Comment (isaacl) [inline src/display/font_loader.js:447]: This is an odd way to increment a for loop.

Comment (calixteman) [reply src/display/font_loader.js:447]: If you think we can do better, please file an issue and write a PR, thank you.
Just for your information, each command is followed by its arguments where their length depends on the command.
So here, after  `i++`, `i` is the position for the first argument, and then each `case` will increment it depending on the number of arguments.

Comment (isaacl) [reply src/display/font_loader.js:447]: Apologies -- I didn't realize `i` was also being incremented in the switch block. I withdraw my comment.

**Pull request merged**